quipucords/scanner/network/connect.py

```diff
@@ -254,18 +266,21 @@ def _handle_ssh_passphrase(credential):
             pass
 
 
-def construct_connect_inventory(hosts, credential, connection_port):
+# pylint: disable=dangerous-default-value
+def construct_connect_inventory(hosts, credential, connection_port,
+                                exclude_hosts=[]):
     """Create a dictionary inventory for Ansible to execute with.
 
     :param hosts: The collection of hosts to test connections
     :param credential: The credential used for connections
     :param connection_port: The connection port
-    :returns: A dictionary of the ansible invetory
+    :param exclude_hosts: Optional. Hosts to exclude test connections
+    :returns: A dictionary of the ansible inventory
     """
     inventory = None
     hosts_dict = {}
 
-    for host in hosts:
+    for host in [host for host in hosts if host not in exclude_hosts]:
```

Comment:
Use set subtraction here too.  However, do we need this if we performed the set difference in the constructor?

Reply:
Yes, because this function is defined outside the scope of the class with the constructor.